### DISCUSSION TITLE
Bed leveling not working at all in Ender 3 S1.

### DISCUSSION THREAD

**Discussion by Victordiazbis** [mriscoc/Ender3V2S1#164] [Q&A]
I have tried both UBL and ABL firmware versions, and in both cases auto mesh generation is a disaster. This is also applicable to bed tramming. The M48 test shows very good results, but the generated meshes have no sense at all as you can see in the picture. I can repeat the process and obtain similar results. Obviously those crazy waves aren't real  , and it's impossible to print anything with auto bed leveling activated. I tried with HS mode deactivated, but nothing changes.
![IMG20220423200452](https://user-images.githubusercontent.com/104123136/169661508-89d20634-13d6-4e61-96b3-30dad8b079c7.jpg)

**Comment by Nbrooks639** [comment]
bed tramming does not get u perfectly level, it gets you close, you are deff. going to have a variation of numbers with that many points, also it looks like your bed is warped, the left side is perfect, just play a little bit with the right side and you will be good. i have seen a lot worse and prints come out just fine.

**Comment by Victordiazbis** [comment]
The bed is manually trammed, because the auto tramming produces a tilted bed.  The bed is not warped. Actually the manual mesh that produces good prints is as follows. 
![IMG20220520125746](https://user-images.githubusercontent.com/104123136/169677329-e2a8647c-6da0-4aaf-84b2-76f13caef83f.jpg)
I know the bed level is not perfect, but no matter how well I level the corners, the mesh does not reflect the geometry of the bed. It's very strange, but I can perform an assisted auto tramming and get 0.0 in the four corners, but when I check with a paper sheet, immediately after at the exact same locations, it is clear that the  bed is tilted. As summary, the CR touch measures are more or less repetitive, but they don't reflect the real bed geometry. 🤔

**Reply by Victordiazbis** [reply]
The source of the problem is already detected. Many thanks Miguel!!!!. One wheel of x axis was not perfectly round. In consequence the extruder was oscillating in its displacement along X axis. The defect was almost undetectable to naked eye, and didn't affect to print quality. However it made the autolevel completely useless. 

Lessons learnt: Care about the kinematics of the machine, and special attention to the wheels. They are the foundations to build a good printer. Otherwise the experience can be ruined even with Miguel's fantastic firmware
